main/main.cpp

```diff
@@ -2836,6 +2878,13 @@ bool Main::iteration() {
 		Input::get_singleton()->flush_buffered_events();
 	}
 
+	if (movie_writer) {
+		RID main_vp_rid = RenderingServer::get_singleton()->viewport_find_from_screen_attachment(DisplayServer::MAIN_WINDOW_ID);
```

Comment:
The main viewport's `transparent_bg` is false as default, so it needs to be set to true.

```suggestion
		RID main_vp_rid = RenderingServer::get_singleton()->viewport_find_from_screen_attachment(DisplayServer::MAIN_WINDOW_ID);
		RenderingServer::get_singleton()->viewport_set_transparent_background(main_vp_rid, true);
```

But maybe there is a more appropriate place to do this outside of `iteration()`.

Reply:
That is entirely up to you, it won't happen by default.

Reply:
At some point we should fix transparent windows in Godot (with Vulkan its kind of hit or miss) then with that option I guess it can work out of the box.

Reply:
In addition to that, you must always avoid blending the background color as a matte behind a transparent object.

For example, there is already a problem with MaterialEditor. The following image is not shaded and should be the same color as the Color parameter, but clearly an implicit background color is blended in.

![image](https://user-images.githubusercontent.com/61938263/174400622-ec016c71-c094-4393-8202-5369d51f5472.png)

When blending alphas, they should be neither a premultiplied alpha nor a matte alpha, but basically a straight alpha.

https://limnu.com/webgl-blending-youre-probably-wrong/

This can be sent as a separate Proposal from TransparentWindow as well.